feat(docs): add guide for ButterCMS

jakelumetta opened the pull request:
## Description

Adds a comprehensive guide for using Gatsby with ButterCMS and related changes.  I think I covered all the updates I need to make (i.e. updating the nav links) but if I missed any please let me know.  I also added an entry + link to our Gatsby Starter project and source plugin.

Looking forward to your feedback!

## Related Issues

https://github.com/gatsbyjs/gatsby/issues/9757

RobinCsl commented on docs/docs/headless-cms.md:
```diff
@@ -17,8 +17,8 @@ The guides in this section will walk through the process of setting up content s
 
-- [Cosmic JS](https://www.gatsbyjs.org/packages/gatsby-source-cosmicjs)
 - [ButterCMS](https://www.gatsbyjs.org/packages/gatsby-source-buttercms)
+- [Cosmic JS](https://www.gatsbyjs.org/packages/gatsby-source-cosmicjs)
```
What's the reason behind re-ordering these entries?

jakelumetta replied:
Alphabetical? :)

calcsam replied:
There's actually precedent for this! One reason that "Amazon" was chosen as the name for Jeff Bezos' new company was that many link directories at the time, such as Yahoo, were alphabetical. "Abracadabra" was also considered.

jakelumetta replied:
@calcsam Good call.  Want me to update the list to be alphabetical?

calcsam replied:
No, though thanks for the offer! We're going to order based on what is popular as measured by downloads, as well as CMS size / adoption:

https://github.com/gatsbyjs/gatsby/pull/11254

calcsam commented on www/src/data/sidebars/doc-links.yaml:
```diff
@@ -92,6 +92,8 @@
           link: /docs/headless-cms/
           items:
+            - title: Sourcing from ButterCMS
+              link: /docs/sourcing-from-buttercms/
```
We're trying to keep the "above-the-fold" links to 5, as linked by the most popular source plugins in https://www.gatsbyjs.org/plugins/?=gatsby-source- (with a bump to WP and Drupal since they're globally popular CMSs). So unfortunately, we can't accept this link. 

Instead, you should add it to the "Other CMS systems" section -- you can change to `ButterCMS (docs, guide, starter)`

jakelumetta replied:
@calcsam ok thanks

jakelumetta commented:
@Robdel12 thanks for the heads up on yarn formatting. I ran that command but it seems to have updated more files than just mine?  

https://cl.ly/3add8d7474bf

I'm happy to commit these but don't want to mess up other files? Thoughts?

jakelumetta commented:
@calcsam @Robdel12 - Mind taking a look at the build error? Sorry, I'm not sure I follow what's going on.  I think I've addressed all of your feedback - let me know if there's anything else I can do.

Cheers

RobinCsl commented:
I think you mis-tagged me, @jakelumetta. I'm @RobinCsl 😄 You have some conflicts to solve first.

jakelumetta commented:
Ha - sorry!  Ok I've resolved the merge conflicts.

DSchau commented:
@jakelumetta just fixed the issues. To be clear, you only edited the two files (manually) that are in this PR currently, correct?

```
docs/docs/sourcing-from-buttercms.md
docs/starters.yml
```

If so--this is probably good to go. If you have a second, I'd be interested in hearing your workflow and _how_ we got all those changes. It seems like you were using a newer version of `prettier` than we use. Did you perhaps use `npm install` to install dependencies instead of `yarn`?

jakelumetta commented:
@DSchau Ah yes, you nailed it. Sorry - I used `npm install` in the repo then ran the yarn command.

Thanks for the review! 🍻

jakelumetta commented:
Hey @DSchau  - I just noticed this page:  https://www.gatsbyjs.com/guides/  

Can I add an entry for the ButterCMS guide in a new PR there?

wardpeet commented on docs/starters.yml:
```diff
@@ -1685,6 +1685,21 @@
     - Documented to get you set up with Craft CMS quickly
     - Code referenced in repo
+- url: https://gatsby-starter-buttercms.netlify.com/
+  repo: https://github.com/ButterCMS/gatsby-starter-buttercms
```
I have a few comments on the starter itself.

You have a Readme.md that says
```
.gitignore: This file tells git which files it should not track / not maintain a version history for.

.prettierrc: This is a configuration file for Prettier. Prettier is a tool to help keep the formatting of your code consistent.
```
but I can't seem to find a gitignore or prettier file, could you add it or remove these lines?

It would also be handy if you add https://gatsby-starter-buttercms.netlify.com/ to your README so people can easily view it from the github page.

wardpeet commented on docs/starters.yml:
```diff
@@ -1685,6 +1685,21 @@
+  description: A starter template for spinning up a Gatsby+ ButterCMS site
+  tags:
+    - Blog
+    - FAQ
```
```suggestion
```

We don't have a FAQ tag yet and it's not really something we want to add as a tag

wardpeet commented on docs/starters.yml:
```diff
@@ -1685,6 +1685,21 @@
+    - Blog
+    - FAQ
+    - SEO
+    - Marketing Pages
```
```suggestion
```
Same as above we don't have this as a tag

wardpeet commented on docs/starters.yml:
```diff
@@ -1685,6 +1685,21 @@
+    - FAQ
+    - SEO
+    - Marketing Pages
+    - Marketing Site
```
```suggestion
```
Same as above we don't have this as a tag

wardpeet commented on docs/starters.yml:
```diff
@@ -1685,6 +1685,21 @@
+- url: https://gatsby-starter-buttercms.netlify.com/
+  repo: https://github.com/ButterCMS/gatsby-starter-buttercms
+  description: A starter template for spinning up a Gatsby+ ButterCMS site
+  tags:
```
You could add a tag
`ButterCMS`

jakelumetta commented:
@wardpeet Also for https://www.gatsbyjs.com/guides/ - who's the best person to ask about that.  It would be helpful to know the goal of that page; many headless CMS's are listed there and I don't see why there would be a limit on the number of guides you want to show off.  I'd love to add in ButterCMS as we're making a major push to support Gatsby (as I hope you can tell 😄 ).

calcsam commented:
I'll email you @jakelumetta

jakelumetta commented:
🎉
